presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -433,7 +435,15 @@ public StreamProperties visitEnforceSingleRow(EnforceSingleRowNode node, List<St
         @Override
         public StreamProperties visitAssignUniqueId(AssignUniqueId node, List<StreamProperties> inputProperties)
         {
-            return Iterables.getOnlyElement(inputProperties);
+            StreamProperties sourceProperties = Iterables.getOnlyElement(inputProperties);
+            return new StreamProperties(
+                    sourceProperties.getDistribution(),
+                    Optional.of(node.getOutputSymbols()),
+                    Optional.of(new OrderingScheme(
```

Comment:
@sopel39 @martint 

`OrderingScheme` class contains two fields: `orderBy` and `orderings`. What's the difference between them? Why do we need `orderBy`? Isn't `orderBy` same as `orderings.keys()`?

```
public class OrderingScheme
{
    private final List<Symbol> orderBy;
    private final Map<Symbol, SortOrder> orderings;
```

Reply:
@mbasmanova 
`orderBy` list preservers order of sort columns

`orderings` map are created via:
```
this.orderings = ImmutableMap.copyOf(orderings);
```
which means `keys()` will be returned in the same order as entries from copied `orderings` map. We shouldn't assume that map that we copy from will return keys in the same order as `orderBy` list

Reply:
@sopel39 Karol, thanks for explaining. It seems more natural to me to use parallel lists: `List<Symbol>` and `List<SortOrder>` or a list of pairs, but I understand why there are two now.